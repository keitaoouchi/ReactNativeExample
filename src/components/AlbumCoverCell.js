import React from "react";
import {Animated, Easing, Text, View} from "react-native";
import Workaround from "../misc/Workaround";

export default class AlbumCoverCell extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      album: props.album,
      fadeAnim: new Animated.Value(0),
    };
  }

  render() {
    return (
      <View style={style.root}>
        <Animated.Image
          style={[style.image, {opacity: this.state.fadeAnim}]}
          source={{uri: this.state.album.coverImageURL}}
          onLoadEnd={() => {
            Animated.timing(
              this.state.fadeAnim,
              {
                toValue: 1,
                easing: Easing.easeInOut,
              },
            ).start();
          }}
        />
        <Animated.View style={[style.mask, {opacity: this.state.fadeAnim}]}>
          <Text
            style={ style.artistName }>{this.state.album.artist.name}</Text>
          <Text style={ style.albumName }>{this.state.album.title}</Text>
        </Animated.View>

      </View>
    );
  }
};

const style = {
  root: {width: Workaround.SCREEN_WIDTH, height: Workaround.SCREEN_WIDTH},
  image: {width: Workaround.SCREEN_WIDTH, height: Workaround.SCREEN_WIDTH},
  mask: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    backgroundColor: 'rgba(0,0,0,0.7)',
    padding: 12,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  artistName: {
    marginBottom: 8,
    color: 'rgba(255,255,255,0.5)',
  },
  albumName: {
    color: 'white',
  },
};
