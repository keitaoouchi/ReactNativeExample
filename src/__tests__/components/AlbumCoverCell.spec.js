import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import AlbumCoverCell from "../../../src/components/AlbumCoverCell";
import AlbumFactory from "../factories/AlbumFactory";

describe('AlbumCoverCell', () => {
  it('renders correctly', () => {
    const album = AlbumFactory.build('Album');
    const tree = renderer.create(
      <AlbumCoverCell album={album}/>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
