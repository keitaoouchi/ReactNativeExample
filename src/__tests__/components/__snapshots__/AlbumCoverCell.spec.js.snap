// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumCoverCell renders correctly 1`] = `
<View
  style={
    Object {
      "height": 750,
      "width": 750,
    }
  }
>
  <Image
    onLoadEnd={[Function]}
    source={
      Object {
        "uri": "https://example.com/1",
      }
    }
    style={
      Object {
        "height": 750,
        "opacity": 0,
        "width": 750,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.7)",
        "bottom": 0,
        "flex": 1,
        "justifyContent": "center",
        "left": 0,
        "opacity": 0,
        "padding": 12,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgba(255,255,255,0.5)",
          "marginBottom": 8,
        }
      }
    >
      Name-1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      AlbumTitle-1
    </Text>
  </View>
</View>
`;
